mlflow/store/sqlalchemy_store.py

```diff
@@ -227,11 +228,14 @@ def get_metric(self, run_uuid, metric_key):
         return metric.value
 
     def search_runs(self, experiment_ids, search_expressions, run_view_type):
-        raise NotImplementedError()
-
-    def list_run_infos(self, experiment_id, _=None):
-        exp = self.session.query(SqlExperiment).get(experiment_id)
-        infos = []
-        for run in exp.runs:
-            infos.append(run.to_mlflow_entity().info)
-        return infos
+        runs = [self._get_run(r.run_uuid, run_view_type).to_mlflow_entity()
```

Comment:
This seems somewhat inefficient if my understanding is correct:

- list_run_infos gets the entire run data (`run.to_mlflow_entity()` implies this), and then throws out everything but info.
- We then call get_run for every resultant run.

If we had an internal variant of `list_run_infos` that returned the full Run, it seems like we could avoid these extra O(n) database calls. Alternatively, we could have list_run_infos rely on search_runs with `anded_expressions=[]`.

Reply:
Added a private `_list_runs`